### PR TITLE
DOC: add missing docs

### PR DESCRIPTION
**Pull request by mattip**
Some of the top-level `*.rst.txt` docs were not part of our on-line or PDF docs:

- `DISTUTILS.rst.txt`
- `C_STYLE_GUIDE.rst.txt`

Additionally, the ~`conduct/report_handling_manual.rst` and~ `alignment.rst` documents were not linked into a TOC, thus unavailable via links.

The top level `CAPI.rst.txt` is not linked in, I think we can drop or move it to informational NEPS if it has historical relevance still.

Edit: "orphaned" dos that are not in a TOC

### CODE REVIEW

**Comment by rgommers** [comment]
Good idea to clean this up. A few comments:

- It looks to me like the distutils guide should go into the user guide, not the reference docs.
- C style guide should either become a NEP or be part of the dev guide, not the reference docs.
- The distutils section could also use a few simple cleanups, it's clear now that it was written 10 years ago: change references to "SciPy" to general scientific packages, edit the "useful functions ..." section and turn things into links there.

> Additionally, the conduct/report_handling_manual.rst and alignment.rst documents were not linked into a TOC, thus unavailable via links.

That's not quite how it works, e.g. the `report_handling` page is linked from https://www.numpy.org/devdocs/dev/conduct/code_of_conduct.html#incident-reporting-resolution-code-of-conduct-enforcement. I'll comment on these changes separately.

> The top level CAPI.rst.txt is not linked in, I think we can drop or move it to informational NEPS if it has historical relevance still.

That sounds right. Would be good to verify that all content is already present in the docs - did you do that?

File: doc/source/dev/index.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@ Contributing to NumPy
    :maxdepth: 3
 
    conduct/code_of_conduct
+   conduct/report_handling_manual
```
[the remark this answers is not in the document]
Comment (mattip) [reply doc/source/dev/index.rst]: Removed

File: doc/source/reference/internals.rst [hunk 2]
```diff
@@ -5,5 +5,6 @@ NumPy internals
 .. toctree::
 
    internals.code-explanations
+   alignment
```
[the remark this answers is not in the document]
Comment (mattip) [reply doc/source/reference/internals.rst]: Done

**Comment by mattip** [comment]
I broke these::

    - It looks to me like the distutils guide should 
      go into the user guide, not the reference docs.
    - C style guide should either become a NEP or be 
      part of the dev guide, not the reference docs.
    - The distutils section could also use a few simple
      cleanups, it's clear now that it was written 10 years ago: 
      - change references to "SciPy" to general scientific packages
      - edit the "useful functions ..." section and turn things into links there.


out into a separate issue #12261

**Comment by mattip** [comment]
Changesets aeefb51 and 00e8e16 refactored and removed the outdated CAPI.rst.txt, after merging relevant information back into the docs

**Comment by mattip** [comment]
> C style guide should either become a NEP or be part of the dev guide, not the reference docs.

Moved to dev section

**Comment by mattip** [comment]
@rgommers ping

**Comment by rgommers** [comment]
Okay, in it goes. Thanks @mattip

**Pull request merged**